lib/ansible/plugins/connection/ssh.py

```diff
@@ -356,11 +356,11 @@ def _handle_error(remaining_retries, command, return_tuple, no_log, host, displa
 
     # For other errors, no execption is raised so the connection is retried and we only log the messages
     if 1 <= return_tuple[0] <= 254:
-        msg = "Failed to connect to the host via ssh:"
+        msg = u"Failed to connect to the host via ssh:"
         if no_log:
-            msg = '{0} <error censored due to no log>'.format(msg)
+            msg = u'{0} <error censored due to no log>'.format(msg)
```

Comment:
It's possible for this to traceback on Python2 if sshpass was used and we go through the code path here: https://github.com/ansible/ansible/pull/58121/files#diff-2cdbb4c74e2af35eda8ddf643f904b1aR336

Use a text string in that code path to avoid the chance of failure.

Reply:
Actually scratch that... I'm wrong about that because msg is reset just above this.  However, it looks like the stuff in the code path I mentioned above doesn't have any effect (because we're overwriting it in this conditonal block).

Reply:
Hrm, you are right. It looks like we are missing a `raise` after line 340.  I'll look into that in a separate PR at a later date.  I'll log an issue for it so it's not lost.